### PR TITLE
Angular followers frontend for user profiles

### PR DESCRIPTION
**Pull request by henryksloan**
**This PR has been updated to reflect #44**
**The only features not included in this PR now are follow list page selection (currently only shows page 0) and follow count display on mobile**

# Done
This PR adds to #38 a follow button and fully working followers/following lists with dummy data
## info.service.ts
- Added dummy users for use in dummy follow lists
- Added utilities to produce observables and requests for `/follow` API
    - POST new follow
    - DELETE existing follow
    - GET full list of type={following/followers}
    - GET boolean whether or not userId follows otherId
    - **These GETs are currently populated with dummy data, but contain draft code in the comments**
## pages/user-profile/follow-list/follow-entity/*
- Replaces placeholder entities from #38
- Displays the user's profile picture and name
- Encompasses a link to the user's profile

# TODO
- Create a default profile picture implementation that can scalably be included in lists like this (the hashing implementation from `UserProfile` should be fine, assuming we have pagination or low user count)
- Add follower/following count in a mobile-friendly manner
    - Possibly use a different display method for these numbers on mobile
- Ensure follow button and lists display well on mobile

# Screenshots
image

### CODE REVIEW

**Comment by kirantumkur** [comment]
@XLuna24 @TheGuyWhoCodes Please review...

**Comment by TheGuyWhoCodes** [comment]
This looks good. On the PFP do we want to stick with the square icon? With the rounded buttons it looks a little out of place.

**Comment by henryksloan** [comment]
@TheGuyWhoCodes Good point about the PFP shape, I'll do some experimenting. The Angular CI check is still failing on this PR despite rebasing after merging profiles; is there something more I have to do to fix that, or is the CI broken?

File: mediaphile/src/app/info.service.ts [hunk 1]
```diff
@@ -117,19 +126,36 @@ export class InfoService {
...
+
+  public userFollows(userId: string, otherId: string) {
+    // TODO: Remove placeholder
+    return of(otherId === "123"); /* this.http.get(this.followListEndpoint, {
+      params: {
+        "userId": userId,
+        "targetId": otherId // TODO: Ensure these names are correct
+      }
+    }); */
+  }
```
Comment (kirantumkur) [inline mediaphile/src/app/info.service.ts:160]: Do you not want to fix this?

Comment (henryksloan) [reply mediaphile/src/app/info.service.ts:160]: I've been working on that fix for a little while, just making sure it works with Xavier's changes.

**Pull request merged**